crates/doc/src/inference.rs

```diff
@@ -444,6 +445,82 @@ impl ObjShape {
             None
         }
     }
+
+    fn widen<'n, N>(&mut self, fields: &'n N::Fields, loc: Location, is_first_time: bool) -> bool
+    where
+        N: AsNode,
+    {
+        use crate::{Field, Fields};
+        // If a particular location defines `additionalProperties` as a subschema
+        // don't add the field from `AsNode`, instead jump straight to squashing
+        match self.additional.as_mut() {
+            // We only want to squash if your `additionalProperties`
+            // is set to something other than `false`.
+            Some(addl) if !addl.type_.eq(&types::INVALID) => {
+                fields.iter().enumerate().fold(false, |accum, (idx, node)| {
+                    accum || addl.widen(node.value(), loc.push_item(idx))
```

Comment:
this is still incorrect usage -- it should be `push_property(node.property)`, as we're traversing RHS object properties, not RHS array indices.

Reply:
Ah yep. On this topic though, is my comment [here](https://github.com/estuary/flow/pull/1126#discussion_r1276654169) right? I.e, if it really _is_ an array index, I should be using `push_item(index)`, right?

Reply:
That was also the Node::Object match arm.

but yes if it is an array push_item